internal/verifier/compare.go

```diff
@@ -475,63 +510,166 @@ func (verifier *Verifier) getDocumentsCursor(ctx mongo.SessionContext, collectio
 		}
 	} else {
-		findOptions = task.QueryFilter.Partition.GetFindOptions(clusterInfo, verifier.maybeAppendGlobalFilterToPredicates(andPredicates))
+		pqp := task.QueryFilter.Partition.GetQueryParameters(
+			clusterInfo,
+			verifier.maybeAppendGlobalFilterToPredicates(andPredicates),
+		)
+
+		switch verifier.docCompareMethod.QueryFunction() {
+		case DocQueryFunctionFind:
+			findOptions = pqp.ToFindOptions()
+		case DocQueryFunctionAggregate:
+			aggOptions = pqp.ToAggOptions()
+
+			if verifier.docCompareMethod != DocCompareToHashedIndexKey {
+				panic("unknown aggregate compare method: " + verifier.docCompareMethod)
+			}
+
+			for i, el := range aggOptions {
+				if el.Key != "pipeline" {
+					continue
+				}
+
+				aggOptions[i].Value = transformPipelineForToHashedIndexKey(
+					aggOptions[i].Value.(mongo.Pipeline),
+					task,
+				)
+
+				break
+			}
+
+		default:
+			panic("bad doc compare query func: " + verifier.docCompareMethod.QueryFunction())
+		}
+	}
+
+	var cmd bson.D
+
+	switch verifier.docCompareMethod.QueryFunction() {
+	case DocQueryFunctionFind:
+		cmd = append(
+			bson.D{{"find", collection.Name()}},
+			findOptions...,
+		)
+	case DocQueryFunctionAggregate:
+		cmd = append(
+			bson.D{
+				{"aggregate", collection.Name()},
+				{"cursor", bson.D{}},
+			},
+			aggOptions...,
+		)
 	}
+
 	if verifier.readPreference.Mode() != readpref.PrimaryMode {
 		runCommandOptions = runCommandOptions.SetReadPreference(verifier.readPreference)
 		if startAtTs != nil {
+			readConcern := bson.D{
+				{"afterClusterTime", *startAtTs},
+			}
 
 			// We never want to read before the change stream start time,
 			// or for the last generation, the change stream end time.
-			findOptions = append(
-				findOptions,
-				bson.E{"readConcern", bson.D{
-					{"afterClusterTime", *startAtTs},
-				}},
+			cmd = append(
+				cmd,
+				bson.E{"readConcern", readConcern},
 			)
 		}
 	}
-	findCmd := append(bson.D{{"find", collection.Name()}}, findOptions...)
 
 	// Suppress this log for recheck tasks because the list of IDs can be
 	// quite long.
 	if !task.IsRecheck() {
+		extJSON, _ := bson.MarshalExtJSON(cmd, true, false)
+
 		verifier.logger.Debug().
 			Any("task", task.PrimaryKey).
-			Str("findCmd", fmt.Sprintf("%s", findCmd)).
+			Str("cmd", lo.Ternary(
+				extJSON == nil,
+				fmt.Sprintf("%s", cmd),
+				string(extJSON),
+			)).
 			Str("options", fmt.Sprintf("%v", *runCommandOptions)).
-			Msg("getDocuments findCmd.")
+			Msg("getDocuments command.")
 	}
 
-	return collection.Database().RunCommandCursor(ctx, findCmd, runCommandOptions)
+	return collection.Database().RunCommandCursor(ctx, cmd, runCommandOptions)
+}
+
+func transformPipelineForToHashedIndexKey(
+	in mongo.Pipeline,
+	task *VerificationTask,
+) mongo.Pipeline {
+	return append(
+		slices.Clone(in),
+		bson.D{{"$replaceWith", bson.D{
+			// Single-letter field names minimize the document size.
+			{docKeyInCompare, bson.D(lo.Map(
```

Comment:
(nit) It feels weird to me that we use a constant for `docKeyInCompare` (which is a name I find confusing), but bare literal `"h"` and `"s"` for the others.